fhir-server/src/main/java/com/ibm/fhir/server/filter/rest/FHIRRestServletFilter.java

```diff
@@ -262,43 +264,68 @@ private HTTPHandlingPreference computeHandlingPref(ServletRequest request) throw
      * disrupting the existing behavior we use a new returnPreferenceDefault flag
      * to indicate whether or not the default value has been overridden by a
      * client-specified header value.
-     * @param context
-     * @param request
-     * @param handlingPref
+     * @param context the context containing the HTTP headers
      * @throws FHIRException
      */
-    private void computeReturnPref(FHIRRequestContext context, ServletRequest request, HTTPHandlingPreference handlingPref) throws FHIRException {
+    private void computeReturnPref(FHIRRequestContext context) throws FHIRException {
         HTTPReturnPreference returnPref = defaultHttpReturnPref;
         boolean isDefault = true;
-        String returnPrefString = ((HttpServletRequest) request).getHeader(preferHeaderName + ":" + preferReturnHeaderSectionName);
-        if (returnPrefString != null && !returnPrefString.isEmpty()) {
+        String returnPrefString = getHeaderTokenValue(context, preferHeaderName, preferReturnHeaderSectionName);
+        if (returnPrefString != null) {
             try {
                 returnPref = HTTPReturnPreference.from(returnPrefString);
                 isDefault = false;
-                
+
                 if (log.isLoggable(Level.FINE)) {
                     log.fine("Requested return preference = " + returnPref);
                 }
             } catch (IllegalArgumentException e) {
-                String message = "Invalid HTTP return preference passed in header 'Prefer': '" + returnPrefString + "'";
-                if (handlingPref == HTTPHandlingPreference.STRICT) {
-                    throw new FHIRException(message + "; use 'minimal', 'representation' or 'OperationOutcome'.");
+                String message = "Invalid HTTP return preference 'return=" + returnPrefString + "' passed in header 'Prefer'";
+                if (context.getHandlingPreference() == HTTPHandlingPreference.STRICT) {
```

Comment:
think we need to add to the javadoc to make it clear that the handling preference should be set in the context before calling this method?

Reply:
Updated